lib/main.dart

```diff
@@ -76,11 +52,31 @@ class _MyAppState extends State<MyApp> {
     );
   }
 
-  Scaffold _waitForAmplify() {
-    return const Scaffold(
-      body: Center(
-        child: CircularProgressIndicator(),
-      ),
-    );
+  void _configureAmplify() async {
+    try {
+      await Amplify.addPlugin(AmplifyAuthCognito());
+      await Amplify.configure(amplifyconfig);
+      setState(() {
+        _amplifyConfigured = true;
+      });
+    } on AmplifyAlreadyConfiguredException {
+      ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
+        backgroundColor: Colors.red,
+        content: Text(
+          'Tried to reconfigure Amplify; '
+          'this can occur when your app restarts on Android.',
+        ),
+      ));
+    }
+  }
+
+  Widget buildApp(BuildContext context) {
```

Comment:
Let's not use a function for the widget, especially when this is this small. it does not increase readability and acts anti-pattern.